src/main/java/org/radarcns/monitor/Monitors.java

```diff
@@ -17,25 +17,43 @@
 package org.radarcns.monitor;
 
 import com.mongodb.MongoClient;
+import java.io.IOException;
+import java.net.ConnectException;
 import java.util.HashMap;
-import org.radarcns.restapi.source.Source;
-import org.radarcns.restapi.spec.SourceSpecification;
-import org.radarcns.source.SourceCatalog;
+import org.radarcns.catalog.SourceCatalog;
+import org.radarcns.domain.managementportal.SourceType;
+import org.radarcns.domain.restapi.Source;
+import org.radarcns.service.SourceMonitorService;
 
 /**
  * Generic Data Accesss Object database independent.
  */
 public class Monitors {
 
-    /** Map containing actual implementations of each source monitor. **/
-    private final HashMap<String, SourceMonitor> hooks;
+    /** Map containing actual implementations of each sourceType monitor. **/
+    private final HashMap<String, SourceMonitorService> hooks;
 
     /** Singleton instance. **/
-    private static final Monitors INSTANCE = new Monitors();
+    private static final Monitors INSTANCE;
 
     /** Constructor. **/
-    public Monitors() {
+    private Monitors(SourceCatalog catalog) {
```

Comment:
Either fix (actual data lookups) and use `@Inject` instead of singleton pattern, or remove the class altogether.